packages/relay/src/lib/db/repositories/hbarLimiter/ipAddressHbarSpendingPlanRepository.ts

```diff
@@ -102,6 +102,36 @@ export class IPAddressHbarSpendingPlanRepository {
     }
   }
 
+  /**
+   * Deletes all IP address spending plans from the cache.
+   *
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @returns {Promise<void>} - A promise that resolves when all IP address spending plans are deleted.
+   */
+  async deleteAll(requestDetails: RequestDetails): Promise<void> {
+    const key = this.getKey('*');
+    const keys = await this.cache.keys(key, 'deleteAll', requestDetails);
+    for (const key of keys) {
+      await this.cache.delete(key, 'deleteAll', requestDetails);
+    }
+    this.logger.trace(`${requestDetails.formattedRequestId} Deleted all IP address spending plans`);
+  }
+
+  /**
+   * Gets all IP address spending plans from the cache.
+   *
+   * @param {RequestDetails} requestDetails - The request details for logging and tracking.
+   * @returns {Promise<IPAddressHbarSpendingPlan[]>} - A promise that resolves with an array of IP address spending plans.
+   */
+  async getAllPlans(requestDetails: RequestDetails): Promise<IPAddressHbarSpendingPlan[]> {
```

Comment:
To keep this class, `IPAddressHbarSpendingPlanRepository`, and the `EthAddressHbarSpendingPlanRepository` class in sync, let's also add another getAllPlans() method to the `EthAddressHbarSpendingPlanRepository` class.